rollbar/__init__.py

```diff
@@ -536,6 +536,23 @@ def wait(f=None):
         return f()
 
 
+def watch(key, extra_data=None):
```

Comment:
How we decided to use `watch()` We use `scoped()` in other libraries so it'd be nice to be consistent with it

Reply:
We had a meeting with the DX team and the members of this project and we agreed `watch` was a suitable name for this feature. I think it's a bit different from `scoped()` because here we have a mandatory param `key` and this gets inserted in the `key` of the `scope` object which is where the pipeline will be looking for this for our LaunchDarkly Integration.

Reply:
it's there any spec for this that rest of people in the engineering team can read? I'd need to understand what is the thinking behind these decisions.

The thing is not which mandatory params we have or not, but what's the meaning of the function and what's the meaning of "scopes" here. Because they are different than the "scope" concept in the ruby library or javascript library.

Reply:
There is no spec on this. Well, I guess there is this spec which the idea was derived from https://rollbar.quip.com/Yt9jAl50toOf/Tech-Spec-Automated-Feature-Flag-Kill-Switch-v15-Milestone-2#JbBACAATdcS . We first thought this would be specific to feature flags, but then decided to take a more general approach, because we saw there was more possibilities for it.

I agree this is different than the "scope" concept in the ruby library (the javascript library deprecated scope), but it's okay to be different (especially for a beta release where we can test the effectiveness and then still update the interface)? We want this method to define a scope but also pass in the `key` with it which we can then look for in the pipeline and then look for and do some actions based on it. If we did scope similar to the ruby library, the user can then pass in a dictionary with the wrong key and the action on the pipeline wouldn't be taken.

Would it be better to have the developer pass in an arbitrary dictionary and have them possibly not set the right key or a method designated to be used with this integration we have?

If you're interested, the workaround to use this method with the rollbar-gem would be something like this:
```
while job
  # write scope object
  scope = {
    :scope => {:key => "your_feature_flag_key"}
  }

  Rollbar.scoped(scope) do
    begin
      job.do_work
    rescue => e
      # Sends a report with the correct scope and key set to feature flag key
      Rollbar.critical(e)
    end
  end

  job = next_job
end
```
I'm guessing this is what you are expecting for the Pyrollbar implementation?